server/apps/publish/archive_publish.py

```diff
@@ -670,6 +671,11 @@ def __init__(self, datasource=None, backend=None):
         super().__init__(datasource=datasource, backend=backend)
 
     def on_update(self, updates, original):
+        # check if we are trying to kill and item that is contained in normal non takes package
+        if original[ITEM_TYPE] != CONTENT_TYPE.COMPOSITE and original.get(LINKED_IN_PACKAGES, None) \
```

Comment:
Other than error message both seem to be doing the same thing. Can we extract it to a function?